kong/plugins/prometheus/exporter.lua

```diff
@@ -420,6 +421,7 @@ local function metric_data(write_fn)
     -- upstream targets accessible?
     local upstreams_dict = get_all_upstreams()
     for key, upstream_id in pairs(upstreams_dict) do
+      yield(true)
```

Comment:
Please refer to the implement of `yield()` in `tools/utils.lua`.
Perhaps we should use `yield(false)` and pass the cached `phase` param for performance.

Reply:
`yield(true)` was used to avoid excessive yielding.

Reply:
Adding the second param `phase` should be a good idea for performance.

Reply:
`access` phase?